AppCenterDataStorage/AppCenterDataStorage/MSDataSourceError.h

```diff
@@ -3,7 +3,7 @@
 
 #import <Foundation/Foundation.h>
 
-@interface MSDataSourceError : NSObject
+@interface MSDataSourceError : NSError
```

Comment:
curious, why inheriting from NSError?

Reply:
Because it's an error?  It still inherits from NSObject via NSError

Reply:
You need to remove some of properties then.

Reply:
if it's an error, then NSError property is not needed. 
I still prefer to keep it as NSObject

Reply:
Are you going to keep it or change back to `NSObject`? I need to know this change for my implementation.

Reply:
Reverted.